### PR TITLE
Show instance vars in `set_pipeline` step header

### PR DESCRIPTION
**Pull request by aoldershaw**
[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## What does this PR accomplish?

**Bug Fix** | Feature | Documentation

Screen Shot 2021-01-08 at 3 29 32 PM

## Changes proposed by this PR:

* Refactor `StepTree.Models.Step` to include `Concourse.BuildStep` - this allows us to be more flexible in what we put in the step headers, as we can use arbitrary data from the public plan for each step
* Display instance vars as key-value pairs (just as we render the `across` step's key-value pairs

Screen Shot 2021-01-08 at 3 32 15 PM

## Contributor Checklist

- [x] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [x] [Signed] all commits
- [x] Added tests (Unit and/or Integration)
- [ ] Updated [Documentation]
- [ ] Added release note (Optional)

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

**Comment by aoldershaw** [comment]
@matthewpereira unrelated to this PR, but I was thinking this could be a way to display instanced pipelines in the sidebar, which was something I wasn't sure about re: favouriting instances directly (rather than just favouriting instance groups)

**Comment by matthewpereira** [comment]
This looks good to me. Great work, @aoldershaw ! Let's try it and see what kind of feedback we get. :shipit:

**Review by chenbh — COMMENTED** [review]
One thing I'm confused about is that we have `viewStepState` which used for normal steps, and `viewStepStateWithoutTooltip` which is exactly what it sounds like but is only used for across steps. 

Is there a reason that we're duplicating so much code between those 2 functions? https://github.com/concourse/concourse/blob/e40f97bfefa66e430f8dc7fe5753d33f9d02cf37/web/elm/src/Build/StepTree/StepTree.elm#L990-L1139

File: web/elm/src/Build/StepTree/StepTree.elm [hunk 1]
```diff
@@ -1142,19 +1139,156 @@ viewStepState state stepID =
...
-                    ]
+        headerWithContent label changedTooltip content =
+            Html.div
+                [ style "display" "flex" ]
+                [ viewStepHeaderLabel label changedTooltip step.changed step.id
+                , Html.h3 [ style "display" "flex" ] content
+                ]
+
+        simpleHeader label changedTooltip name =
+            headerWithContent label changedTooltip [ Html.text name ]
```
[the remark this answers is not in the document]
Comment (aoldershaw) [reply web/elm/src/Build/StepTree/StepTree.elm:1153]: yeah that's equivalent. it's the same syntax as "regular" functions (i.e. those outside of a `let`), just with the type signature omitted (which is also valid for top-level functions, but the linter will complain)

**Comment by aoldershaw** [comment]
> One thing I'm confused about is that we have viewStepState which used for normal steps, and viewStepStateWithoutTooltip which is exactly what it sounds like but is only used for across steps.
>
> Is there a reason that we're duplicating so much code between those 2 functions?

@chenbh huh, not sure why I did that, doesn't look like there's any reason for it. Simplified in 6d70c6edd284428a25d4e7e9d6d80cad7d2fac62

**Review by chenbh — APPROVED** [review]
kicking off the tests again since it looks like ci was redeployed in the middle of it

**Pull request merged**